File: problems/sieveOfEratosthenes/javascript/sieveOfEratosthenes.js
function sieveOfEratosthenes(max) {
  const boolean_array = new Array(max + 1);
  boolean_array[0] = false;
  boolean_array[1] = false;

  for (let i = 2; i <= max; i++) {
    boolean_array[i] = true;
  }

  let primes = [];
  for (let curr = 2; curr <= max; curr++) {
    if (boolean_array[curr]) {
      // The current number is prime!
      primes.push(curr);

      // Now mark all multiples as NOT prime (i.e. false)
      for (let multiple = curr * 2; multiple <= max; multiple += curr) {
        boolean_array[multiple] = false;
      }
    }
  }

  return primes;
}

module.exports = sieveOfEratosthenes;

File: problems/sieveOfEratosthenes/swift/sieveOfEratosthenes.swift
func sieveOfEratosthenes(for max: Int) -> [Int] {
  var bool_array: [Bool] = Array(repeating: true, count: max + 1)
  bool_array[0] = false;
  bool_array[1] = false;

  var primes: [Int] = []

  for num in stride(from: 2, through: max, by: 1) {
    if bool_array[num] == true {
      primes.append(num)

      var multiple = num * 2
      while (multiple <= max) {
        bool_array[multiple] = false
        multiple *= num
      }
    }
  }

  return primes
}